src/store/singing.ts

```diff
@@ -1717,6 +1717,128 @@ export const singingStore = createPartialStore<SingingStoreTypes>({
+      async ({ dispatch }, { filePath }: { filePath?: string }) => {
+        // USTファイルの読み込み
+        if (!filePath) {
+          filePath = await window.backend.showImportFileDialog({
+            title: "UST読み込み",
+            name: "UST",
+            extensions: ["ust"],
+          });
+          if (!filePath) return;
+        }
+        // ファイルの読み込み
+        const fileData = getValueOrThrow(
+          await window.backend.readFile({ filePath })
+        );
+
+        // ファイルフォーマットに応じてエンコーディングを変える
+        // UTF-8とShiftJISの2種類に対応
+        let ustData;
+        try {
+          ustData = new TextDecoder("utf-8").decode(fileData);
+          // ShiftJISの場合はShiftJISでデコードし直す
+          if (ustData.includes("\ufffd")) {
+            ustData = new TextDecoder("shift-jis").decode(fileData);
+          }
+        } catch (error) {
+          throw new Error("Failed to decode UST file.");
+        }
+        if (!ustData || typeof ustData !== "string") {
+          throw new Error("Failed to decode UST file.");
+        }
+
+        // 初期化
+        const tpqn = DEFAULT_TPQN;
+        const tempos: Tempo[] = [
+          {
+            position: 0,
+            bpm: DEFAULT_BPM,
+          },
+        ];
+        const timeSignatures: TimeSignature[] = [
+          {
+            measureNumber: 1,
+            beats: DEFAULT_BEATS,
+            beatType: DEFAULT_BEAT_TYPE,
+          },
+        ];
+        const notes: Note[] = [];
+
+        // USTファイルのセクションをパース
+        const parseSection = (section: string): { [key: string]: string } => {
+          const sectionNameMatch = section.match(/\[(.+)\]/);
+          if (!sectionNameMatch) {
+            throw new Error("Failed to parse UST file.");
+          }
+          const params = section.split(/[\r\n]+/).reduce((acc, line) => {
+            const [key, value] = line.split("=");
+            if (key && value) {
+              acc[key] = value;
+            }
+            return acc;
+          }, {} as { [key: string]: string });
+          return {
+            ...params,
+            sectionName: sectionNameMatch[1],
+          };
+        };
+
+        try {
+          // セクションを分割
+          const sections = ustData.split(/^(?=\[)/m);
+          // ポジション
+          let position = 0;
+          // セクションごとに処理
+          sections.forEach((section) => {
+            const params = parseSection(section);
+            // SETTINGセクション
+            if (params.sectionName === "#SETTING") {
+              const tempo = Number(params["Tempo"]);
+              if (tempo) tempos[0].bpm = tempo;
+            }
+            // ノートセクション
+            if (params.sectionName.match(/^#\d{4}/)) {
+              // テンポ変更があれば追加
+              const tempo = Number(params["Tempo"]);
+              if (tempo) tempos.push({ position, bpm: tempo });
+              const noteNumber = Number(params["NoteNum"]);
+              const duration = Number(params["Length"]);
+              const lyric = params["Lyric"];
```

Comment:
適当に試したUSTだと、ここが「a か」みたいになってました。
ファイルフォーマットの公式ドキュメントが特に見当たらないので分かりませんが、パッとググった感じ前の音の連続音が書かれてる時がありそう･･････？

こうしておけばうまいこと行きました。
```suggestion
              // 歌詞の前に連続音が含まれている場合は除去
              const lyric = params["Lyric"].includes(" ")
                ? params["Lyric"].split(" ")[1]
                : params["Lyric"];
```

「R」も「- R」みたいになってたので、ここで良いはず。
コメントはあってるかどうか自信ないです。。

Reply:
@Hiroshiba 
ありがとうございます！こちらもコミットします！

USTファイルの仕様がよくわからないのですが、いったん以下参考にしております…！
(いまいちよくわからず…)
https://w.atwiki.jp/utaou/pages/64.html

> Lyric　詞
> 書式 : Lyric=文字列
> 定義域 : 例外を除く全ての文字列
> デフォルト値 : 直後の音符のLyric(無ければ空白)
> 例外として、改行、セクション名、エントリ名+「=」を含んだ詞は指定できません。
> 例えば、「あ [#INSERT]」、「あ PreUtterance=」、「あ $foo=bar」
> のようなものです。
> ただし、GUI側の操作などで既にそのような詞になっているときは、
> プラグイン側で変更しない限りそのままです。